src/Build/BuildCheck/Infrastructure/BuildCheckManagerProvider.cs

```diff
@@ -620,15 +628,10 @@ public void StartProjectRequest(ICheckContext checkContext, string projectFullPa
         /// <param name="newImportedProjectFile">The path of the newly imported project file.</param>
         private void PropagateImport(int evaluationId, string originalProjectFile, string newImportedProjectFile)
         {
-            if (_deferredProjectEvalIdToImportedProjects.TryGetValue(evaluationId, out HashSet<string>? importedProjects))
+            if (_deferredProjectEvalIdToImportedProjects.TryGetValue(evaluationId,
+                    out HashSet<string>? importedProjects))
             {
-                lock (importedProjects)
-                {
-                    if (importedProjects.Contains(originalProjectFile))
-                    {
-                        importedProjects.Add(newImportedProjectFile);
-                    }
-                }
+                importedProjects.Add(newImportedProjectFile);
```

Comment:
Why are we removing the `if (importedProjects.Contains(originalProjectFile))` check please?
If we always add to the imported projects, we don't need the originalProjectsFile and the method almost loses it's meaning.

Reply:
The method had just quite confusing pre-existing naming and comment - I've fixed that.

In short - we are just adding to a lookup of all imported logic, that is keyed by the project file that is importing the logic